pkg/query-service/querycache/query_range_cache_test.go

```diff
@@ -232,6 +232,331 @@ func TestFindMissingTimeRanges(t *testing.T) {
+				{Start: 1738749600000, End: 1738836000000}, // 05 Feb 2025 10:00:00 - 06 Feb 2025 10:00:00
+			},
+		},
+		{
+			name:           "Cached intervals covering some parts with gaps",
+			requestedStart: 1738404000000, // 01 Feb 2025 10:00:00
+			requestedEnd:   1738490400000, // 02 Feb 2025 10:00:00
+			step:           60,
+			cacheKey:       "testKey7",
+			cachedData: []querycache.CachedSeriesData{
+				{Start: 1738404000000, End: 1738418400000, Data: []*v3.Series{}}, // 01 Feb 2025 10:00:00 - 14:00:00
+				{Start: 1738425600000, End: 1738432800000, Data: []*v3.Series{}}, // 01 Feb 2025 16:00:00 - 18:00:00
+				{Start: 1738440000000, End: 1738447200000, Data: []*v3.Series{}}, // 01 Feb 2025 20:00:00 - 22:00:00
+				{Start: 1738454400000, End: 1738461600000, Data: []*v3.Series{}}, // 02 Feb 2025 00:00:00 - 02:00:00
+			},
+			expectedMiss: []querycache.MissInterval{
+				// {Start: 1738404000000, End: 1738404060000}, // 01 Feb 2025 10:00:00 - 10:01:00
+				{Start: 1738418400000, End: 1738425600000}, // 01 Feb 2025 14:00:00 - 16:00:00
+				{Start: 1738432800000, End: 1738440000000}, // 01 Feb 2025 18:00:00 - 20:00:00
+				{Start: 1738447200000, End: 1738454400000}, // 01 Feb 2025 22:00:00 - 02 Feb 2025 00:00:00
+				{Start: 1738461600000, End: 1738490400000}, // 02 Feb 2025 02:00:00 - 10:00:00
+			},
+		},
+		{
+			name:           "Non-overlapping cached intervals outside requested range",
+			requestedStart: 1738490400000, // 02 Feb 2025 10:00:00
+			requestedEnd:   1738576800000, // 03 Feb 2025 10:00:00
+			step:           60,
+			cacheKey:       "testKey8",
+			cachedData: []querycache.CachedSeriesData{
+				{Start: 1738404000000, End: 1738447200000, Data: []*v3.Series{}}, // 01 Feb 2025 10:00:00 - 22:00:00
+				{Start: 1738620000000, End: 1738663200000, Data: []*v3.Series{}}, // 03 Feb 2025 22:00:00 - 04 Feb 2025 10:00:00
+			},
+			expectedMiss: []querycache.MissInterval{
+				{Start: 1738490400000, End: 1738576800000}, // 02 Feb 2025 10:00:00 - 03 Feb 2025 10:00:00
+			},
+		},
+		{
+			name:           "No cached data at all",
+			requestedStart: 1738404000000, // 01 Feb 2025 10:00:00
+			requestedEnd:   1738490400000, // 02 Feb 2025 10:00:00
+			step:           60,
+			cacheKey:       "testKey10",
+			cachedData:     nil,
+			expectedMiss: []querycache.MissInterval{
+				{Start: 1738404000000, End: 1738490400000}, // 01 Feb 2025 10:00:00 - 02 Feb 2025 10:00:00
+			},
+		},
+		{
+			name:           "Cached intervals with overlapping and non-overlapping mix",
+			requestedStart: 1738404000000, // 01 Feb 2025 10:00:00
+			requestedEnd:   1738407600000, // 01 Feb 2025 11:00:00
+			step:           60,
+			cacheKey:       "testKey11",
+			cachedData: []querycache.CachedSeriesData{
+				{Start: 1738404000000, End: 1738404600000, Data: []*v3.Series{}}, // 01 Feb 2025 10:00:00 - 10:10:00
+				{Start: 1738404300000, End: 1738405200000, Data: []*v3.Series{}}, // 01 Feb 2025 10:05:00 - 10:20:00 (overlaps with previous)
```

Comment:
This is not possible, it would get merged with `01 Feb 2025 10:00:00 - 10:10:00` and there will be one time window for `10:00:00 - 10:20:00`